frame/nomination-pools/src/lib.rs

```diff
@@ -0,0 +1,1525 @@
+		/// [`Self::withdraw_unbonded_pool`] can be called to try and minimize unlocking chunks.
+		#[pallet::weight(666)]
+		pub fn unbond_other(origin: OriginFor<T>, target: T::AccountId) -> DispatchResult {
+			let caller = ensure_signed(origin)?;
+			let delegator = Delegators::<T>::get(&target).ok_or(Error::<T>::DelegatorNotFound)?;
+			let mut bonded_pool = BondedPool::<T>::get(&delegator.pool)
+				.defensive_ok_or_else(|| Error::<T>::PoolNotFound)?;
+			bonded_pool.ok_to_unbond_other_with(&caller, &target, &delegator)?;
+
+			// Claim the the payout prior to unbonding. Once the user is unbonding their points
+			// no longer exist in the bonded pool and thus they can no longer claim their payouts.
+			// It is not strictly necessary to claim the rewards, but we do it here for UX.
+			Self::do_reward_payout(target.clone(), delegator, &bonded_pool)?;
+
+			// Re-fetch the delegator because they where updated by `do_reward_payout`.
+			let mut delegator =
+				Delegators::<T>::get(&target).ok_or(Error::<T>::DelegatorNotFound)?;
+			// Note that we lazily create the unbonding pools here if they don't already exist
+			let sub_pools = SubPoolsStorage::<T>::get(&delegator.pool).unwrap_or_default();
+			let current_era = T::StakingInterface::current_era().unwrap_or(Zero::zero());
+
+			let balance_to_unbond = bonded_pool.balance_to_unbond(delegator.points);
+
+			// Update the bonded pool. Note that we must do this *after* calculating the balance
+			// to unbond so we have the correct points for the balance:share ratio.
+			bonded_pool.points = bonded_pool.points.saturating_sub(delegator.points);
+
+			// Unbond in the actual underlying pool
+			T::StakingInterface::unbond(delegator.pool.clone(), balance_to_unbond)?;
+
+			// Merge any older pools into the general, era agnostic unbond pool. Note that we do
+			// this before inserting to ensure we don't go over the max unbonding pools.
+			let mut sub_pools = sub_pools.maybe_merge_pools(current_era);
+
+			// Update the unbond pool associated with the current era with the
+			// unbonded funds. Note that we lazily create the unbond pool if it
+			// does not yet exist.
+			sub_pools.unchecked_with_era_get_or_make(current_era).issue(balance_to_unbond);
+
+			delegator.unbonding_era = Some(current_era);
+
+			Self::deposit_event(Event::<T>::Unbonded {
+				delegator: target.clone(),
+				pool: delegator.pool.clone(),
+				amount: balance_to_unbond,
+			});
+			// Now that we know everything has worked write the items to storage.
+			bonded_pool.put();
+			SubPoolsStorage::insert(&delegator.pool, sub_pools);
+			Delegators::insert(target, delegator);
+
+			Ok(())
+		}
+
+		/// Call `withdraw_unbonded` for the pools account. This call can be made by any account.
+		///
+		/// This is useful if their are too many unlocking chunks to unbond, and some can be cleared
+		/// by withdrawing.
+		#[pallet::weight(666)]
+		pub fn pool_withdraw_unbonded(
```

Comment:
What's the scenario around this call?

Reply:
Its for if the pool is maxxed out on unlocking chunks but we don't meet the criteria for permissionleslly  calling `withdraw_unbonded_other`, this can help remove some chunks

Reply:
So this would help if a lot of unbonded delegators are slow to withdraw